src/app/councillors/[contactSlug]/components/AgendaItemResults.tsx

```diff
@@ -42,6 +41,156 @@ const AgendaItemCard = memo(function AgendaItemCard({
   );
 });
 
+function usePagination<T>(items: T[], itemsPerPage: number = 10) {
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const paginationData = useMemo(() => {
+    const totalItems = items.length;
+    const totalPages = Math.ceil(totalItems / itemsPerPage);
+    const startIndex = (currentPage - 1) * itemsPerPage;
+    const endIndex = startIndex + itemsPerPage;
+    const currentItems = items.slice(startIndex, endIndex);
+
+    return {
+      currentItems,
+      totalPages,
+      totalItems,
+      startIndex: startIndex + 1,
+      endIndex: Math.min(endIndex, totalItems),
+      hasNextPage: currentPage < totalPages,
+      hasPreviousPage: currentPage > 1,
+    };
+  }, [items, currentPage, itemsPerPage]);
+
+  const goToPage = (page: number) => {
+    const pageNumber = Math.max(1, Math.min(page, paginationData.totalPages));
+    setCurrentPage(pageNumber);
+  };
+
+  const goToNextPage = () => {
+    if (paginationData.hasNextPage) {
+      setCurrentPage((prev) => prev + 1);
+    }
+  };
+
+  const goToPreviousPage = () => {
+    if (paginationData.hasPreviousPage) {
+      setCurrentPage((prev) => prev - 1);
+    }
+  };
+
+  useEffect(() => {
+    setCurrentPage(1);
+  }, [items]);
+
+  return {
+    ...paginationData,
+    currentPage,
+    goToPage,
+    goToNextPage,
+    goToPreviousPage,
+  };
+}
+
+interface PaginationProps {
+  currentPage: number;
+  totalPages: number;
+  onPageChange: (page: number) => void;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+  onNextPage: () => void;
+  onPreviousPage: () => void;
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  currentPage,
+  totalPages,
+  onPageChange,
+  hasNextPage,
+  hasPreviousPage,
+  onNextPage,
+  onPreviousPage,
+}) => {
+  const getVisiblePages = () => {
+    const maxVisible = 5;
+
+    if (totalPages <= maxVisible) {
+      return Array.from({ length: totalPages }, (_, i) => i + 1);
+    }
+
+    const half = Math.floor(maxVisible / 2);
+    let start = Math.max(1, currentPage - half);
+    const end = Math.min(totalPages, start + maxVisible - 1);
+
+    if (end - start + 1 < maxVisible) {
+      start = Math.max(1, end - maxVisible + 1);
+    }
+
+    return Array.from({ length: end - start + 1 }, (_, i) => start + i);
+  };
+
+  const visiblePages = getVisiblePages();
+
+  if (totalPages <= 1) return null;
+
+  return (
+    <nav className="flex items-center justify-center space-x-2 mt-8">
```

Comment:
this needs some bottom margin:

<img width="1179" height="317" alt="image" src="https://github.com/user-attachments/assets/b496dfe1-9bdc-4339-be77-26f591bdc28d" />

```suggestion
    <nav className="flex items-center justify-center space-x-2 mt-8 mb-8">
```